src/main/java/seedu/address/model/entity/worker/Worker.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    public Worker(Name name, Phone phone, String sex, String employmentStatus, Date dateOfBirth, Date dateJoined,
+                  Designation designation, boolean isTestWorker) {
+        if (isTestWorker) {
+            this.workerIdNum = IdentificationNumber.customGenerateId("W", 1);
+        } else {
+            this.workerIdNum = IdentificationNumber.generateNewWorkerId();
+        }
+        this.name = name;
+        this.phone = phone;
+        this.sex = sex;
+        this.employmentStatus = employmentStatus;
+        this.dateOfBirth = dateOfBirth;
+        this.dateJoined = dateJoined;
+        this.designation = designation;
+    }
+
+
+    public IdentificationNumber getWorkerIdNum() {
+        return workerIdNum;
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public Phone getPhone() {
+        return phone;
+    }
+
+    public String getSex() {
+        return sex;
+    }
+
+    public Date getDateOfBirth() {
+        return dateOfBirth;
+    }
+
+    public Date getDateJoined() {
+        return dateJoined;
+    }
+
+    public Designation getDesignation() {
+        return designation;
+    }
+
+    public String getEmploymentStatus() {
+        return employmentStatus;
+    }
+
+    public void setPhone(Phone phone) {
+        this.phone = phone;
+    }
+
+    public void setSex(String sex) {
+        this.sex = sex;
+    }
+
+    public void setDateOfBirth(Date dateOfBirth) {
+        this.dateOfBirth = dateOfBirth;
+    }
+
+    public void setDateJoined(Date dateJoined) {
+        this.dateJoined = dateJoined;
+    }
+
+    public void setDesignation(Designation designation) {
+        this.designation = designation;
+    }
+
+    public void setEmploymentStatus(String employmentStatus) {
+        this.employmentStatus = employmentStatus;
+    }
+
+    /**
+     * Returns true if both workers have the same identity fields.
+     * This defines a weaker notion of equality between two workers.
+     */
+    public boolean isSamePerson(Worker otherWorker) {
+        if (otherWorker == this) {
+            return true;
+        }
+
+        return otherWorker != null
+            && otherWorker.getName().equals(getName())
+            && otherWorker.getSex().equals(getSex())
+            && (otherWorker.getPhone().equals(getPhone()));
+    }
+
+    /**
+     * Returns true if both workers have the same identity and data fields.
+     * This defines a stronger notion of equality between two workers.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Worker)) {
+            return false;
+        }
+
+        Worker otherPerson = (Worker) other;
+        return otherPerson.getName().equals(getName())
+            && otherPerson.getPhone().equals(getPhone())
+            && otherPerson.getSex().equals(getSex());
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Probably need to write a test for all hashCode methods to pass Coveralls. If it's more convenient for you, you can just include a one-liner hashcode test inside the test method for equality. 

For example:
` assertEquals(testWorker.hashCode(), otherWorker.hashCode());`.

Reply:
thank you for suggestion!